scripts/cnode-helper-scripts/blockLog.sh

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC1090,SC2086,SC2281
+
+#Todo: adapt for node names != cnode
+#Todo: fix logRotation lock
+
+# Option A: you can manually point this script to your nodes config.json file
+CONFIG=""  # for example "/home/user/mynode/logs/node0.json"
+
+# Option B: in CNTools environments just let the script determine config and logfile
+if [ -z "$CONFIG" ]; then
+	. "${CNODE_HOME}"/scripts/env
+fi
+
+unset logfile
+if [[ "${CONFIG##*.}" = "json" ]] && [[ -f ${CONFIG} ]]; then
+	errors=0
+	logfile=$(jq -r '.setupScribes[] | select (.scFormat == "ScJson") | .scName' "${CONFIG}")
+	[[ -z "${logfile}" ]] && echo -e "${RED}Error:${NC} Failed to locate json logfile in node configuration file\na setupScribe of format ScJson with extension .json expected" && errors=1
+	ekgport=$(cat $CONFIG | jq .hasEKG)
+	[[ -z "${ekgport}" ]] && echo -e "${RED}Error:${NC} Failed to locate the EKG Port in node configuration file" && $errors=1
+	[[ "$(jq -r .TraceChainSyncClient "${CONFIG}")" != "true" ]] && echo -e "${RED}Error:${NC} In config file please set ${FG_YELLOW}\"TraceChainSyncClient\":\"true\"${NC}" && errors=1
+	[[ "$(jq -r .TraceBlockFetchClient "${CONFIG}")" != "true" ]] && echo -e "${RED}Error:${NC} In config file please set ${FG_YELLOW}\"TraceBlockFetchClient\":\"true\"${NC}" && errors=1
+	[[ "$(jq -r .TraceChainSyncHeaderServer "${CONFIG}")" != "true" ]] && echo -e "${RED}Error:${NC} In config file please set ${FG_YELLOW}\"TraceChainSyncHeaderServer\":\"true\"${NC}" && errors=1
+	[[ $errors -eq 1 ]] && exit 1
+else 
+	echo -e "${RED}Error:${NC} Failed to locate json configuration file" && exit 1
+fi
+
+echo "$(date) parsing ${logfile} ..." | tee ${CNODE_HOME}/logs/blockLog_debug.log
+
+blockHeightPrev=$(curl -s -H 'Accept: application/json' http://127.0.0.1:$(cat $CONFIG | jq .hasEKG)/ | jq -r .cardano.node.metrics.blockNum.int.val)
+blockTimeDeltaSlots=0
+
+getDeltaMS() {
+	echo $(echo "$2 $4" | awk -F '[:, ]' '{print ($1*3600000+$2*60000+$3*1000+$4)-($5*3600000+$6*60000+$7*1000+$8) }')
+}
+
+while true
+do
+	blockHeight=$(curl -s -H 'Accept: application/json' http://127.0.0.1:$(cat $CONFIG | jq .hasEKG)/ | jq -r .cardano.node.metrics.blockNum.int.val)
```

Comment:
This is already done outside of loop:
`ekgport=$(cat $CONFIG | jq .hasEKG)`
No need to waste resources on this jq call at each loop. 
Also, if Option B is utilized, there is no need to grab ekg port as env does this for you. Both host and port is available as ${EKG_HOST}:${EKG_PORT}. 
So code could be changed to:
[[ -z ${EKG_PORT} ]] && EKG_PORT=$(jq .hasEKG $CONFIG)
And then use ${EKG_PORT} everywhere in script.

Reply:
considered this, thanks